src/Orleans/Serialization/TypeUtilities.cs

```diff
@@ -82,7 +82,7 @@ internal static bool IsOrleansShallowCopyable(this Type t)
 
         private static bool IsTypeFieldsShallowCopyable(TypeInfo typeInfo)
         {
-            return typeInfo.GetFields().All(f => !(f.FieldType.IsEquivalentTo(typeInfo)) && IsOrleansShallowCopyable(f.FieldType));
+            return typeInfo.GetFields().All(f => !f.FieldType.GetTypeInfo().IsEquivalentTo(typeInfo.AsType()) && IsOrleansShallowCopyable(f.FieldType));
```

Comment:
But that example you are linking is actually doing Getty peinó and then passing a type (not a typeinfo)  to the IsEquivalentTo method, which is exactly what I'm doing too. BTW, I did this change in my coreclr branch because it did not compile for me, I'm not speculating. This works in full dotNET because TypeInfo inherits from Type, but that's not the case in netcore, and it's why we use the GetTypeInfo extension method and the AsType